ui/form/settings.go

```diff
@@ -43,7 +43,7 @@ func (s *SettingsForm) Validate() error {
 		return errors.NewLocalizedError("error.settings_mandatory_fields")
 	}
 
-	if s.Password != "" {
+	if s.Password != "" && s.Confirmation != "" {
```

Comment:
Well, if you change this condition and I enter a password and nothing in the confirmation field. The comparison between these 2 fields won't happen and then the password will be changed.

Since Firefox doesn't supports `autocomplete="new-password"` and auto-complete the field even with `autocomplete="off"`, it might be better to try to rename the field name in the HTML form instead.